### DISCUSSION TITLE
Import from Google Maps Saved Places

### DISCUSSION THREAD

**Discussion by tautologe** [organicmaps/organicmaps#928] [Q&A] [answered]
Hi, 

I would like to import my Google Maps Saved Places to Organic Maps. TL;DR: it works kind off with tokml, but titles are missing.

First I recognized that you have to use Google Takeout to get your Saved Places machine readable, it will output them as GeoJSON. **This does not include Places you have saved in a custom list**. They are exported as Bookmarks without any helpful metadata. So if you want to export them, you have to manually move them to the default list.

Second it seems that organic maps imports only kml files. So I had to use https://github.com/mapbox/tokml to convert my data to KML. Just install it using `npm install -g tokml` and then run tokml from the commandline. Importing the output file in organic maps works, but the titles are missing, every imported place has its coordinates as title.

When comparing my converted google maps export with an export from organic maps, you can actually see some differences, so it is propably a fixable problem. 

But before tweaking tokml to get a better match, I wanted to ask if there is a simpler way to achieve that? 
Maybe offering a direct import from google takeout's geojson in organic maps might be interesting to make the exodus from google even easier?

**Comment by biodranik** [comment, marked as answer]
Update as of October 2025: Organic Maps supports GeoJSON import. So you can:
1. Export your Google Maps `Saved Places.json` GeoJSON file using https://takeout.google.com/ and choosing *Maps (your places) / Records of your starred places and place reviews.*. Make sure that JSON format is selected.
2. When Google sends you a zip archive, open it and find `Saved Places.json`.
3. Open this file with Organic Maps.

Note that some coordinates may have zeroes and appear in the Atlantic Ocean under Africa. We'll improve it.

**Reply by tautologe** [reply]
Thanks for your reply! 

Meanwhile I found out, that you can preserve the name/title of the locations by running tokml specifying the field name in the source data as a parameter:
```
tokml --name=Title takeout.json > takeout.kml
```
This worked for me. You can now import the takeout.kml and all the names of the locations are preserved.

**Reply by ThibXR** [reply]
Hi! Thank you for the tutorial @biodranik !
I exported and found the Saved Places.json file, but I can't open it with Organic Maps (the app doesn't show up in the list of apps available to open it), and I also can't import it from within the app (Import bookmarks and tracks -> I select the folder with the json file -> it says "0 files were found")

For some context, I'm using a Fairphone 6 with e/OS and still figuring things out 😄 
Thanks!

**Reply by thany** [reply]
@ThibXR 
And this is because the saved places don't actually contain the places you saved - just Google Places links to them. That's the whole thing that makes this difficult. That, and the fact that there are multiple different exports that one can make, exacerbated by the fact that they come in different formats as well (I've seen CSV, KML and JSON).

It's a hot mess for sure, and Google should do better. But they won't, since Takeout is basically their way of complying with EU regulations and so they aren't willing to do more than the absolute legal minimum.

**Reply by biodranik** [reply]
@ThibXR please try to rename the file to `Saved Places.geojson`. It looks like we didn't catch the .json extension yet, right @kirylkaveryn @strump ?

**Reply by ThibXR** [reply]
@thany Thank you so much for all the explanations! I wasn't expecting much from Google tbh, and so I totally get why things get messy when trying to pull anything useful out of this Takeout export...

@biodranik thanks to you as well: after changing the extension, Organic Maps does detect the file, but it only imports unnamed places and - as you mentioned - a bunch of entiers with random coordinates!

On my side, I can only export my saved places as a JSON file through Google Takeout, so I can't even try the .klm option to preserve the locations names. "Luckily", I didn't save as many places as I thought, so I might just recreate them manually!

Thank you again for your time, really appreciate it!

**Reply by biodranik** [reply]
We'll implement a proper import of Google's JSON/GeoJSON data into Organic Maps. Stay tuned.

**Comment by SynapseScribe** [comment]
resurecting this topic.. for a smooth transition from google maps to organic maps, there should be a clear way on how to export/import places

**Reply by thany** [reply]
It seems there are two things in the way for this to make it into some kind of feature:

1. Saved lists in Google Maps cannot be exported outright. They can only be accessed through Takeout, and they will be returned as a list of Google Places URLs, which is only partially helpful. Saved places (like starred places I think), can be accessed through an API.
2. To integrate either option into an app, it requires API access, which is going to cost the author money to implement. Google API's oftenly aren't free. And even the first X calls that *are* free, demand a credit card, which is just stupid. If I were the author, my response would be "fat chance, not happening".

The best thing this could result in, is documentation on how to do it manually. So feel free to write it up, I guess 🙂

**Comment by nelscity** [comment]
For those of you who are so lucky as to be Esri subscribers, I used JSON to Feature and then Layer to KML to transform the Google Takeout export to something Organic Maps could import. In Google Takeout, get both the Saved Places and Labeled Places if you have it.

**Reply by SynapseScribe** [reply]
> For those of you who are so lucky as to be Esri subscribers, I used JSON to Feature and then Layer to KML to transform the Google Takeout export to something Organic Maps could import. In Google Takeout, get both the Saved Places and Labeled Places if you have it.

what do you mean by lucky?

**Comment by 3uzbcqje** [comment]
It's been a few years. We still need this feature. I would really love to stop using Google Maps, but I have hundreds of places pinned and moving them manually is not feasible.

**Reply by denics** [reply]
Nextcloud maps does this

**Reply by thany** [reply]
Are you sure? Remember that Google does not expose an API for this, so please elaborate.

**Comment by matt78704** [comment]
I built a Map Converter app that takes your Google Takeout export of Saved Places and converts it into a clean, well-structured KML file that you can import into Google My Maps or any other mapping application that supports KML.

✅ Convert many lists at once
✅ Handles large lists
✅ Accurate data conversion
✅ No manual work required
✅ You are only charged for places that convert

Conversion price: $0.10 per location, billed only after the conversion succeeds.

Try it now:[ https://mapconverter.online](https://mapconverter.online/)

**Reply by thany** [reply]
I understand API access costs you money, but this is not the solution. It's not really in the spirit of free and opensource software. It does show that it's crappy of Google to only output Places-links for saved lists, instead of including the actual place information.

If you would allow the user to input their own API key, you could make it free to use, which would be closer in spirit to OSS. Still not great, but at least you wouldn't be getting a bollocksing, Google would.

**Reply by SynapseScribe** [reply]
also i don't want to share my places with random dudes on internet :)

**Comment by matt78704** [comment]
@thany  While I get your point about "the spirit of free and opensource software" this wasn't built specifically for users of Organic Maps. I built it for anyone who wants to move large lists of Saved Places from Google Maps to Google My Maps, which supports KML imports. But, it also works for any other app that supports KML imports. I support both free software and free markets. If someone does not want to pay to use my app, that's fine.

@Uj947nXmRqV2nRaWshKtHzTvckUUpD All data conversion happens client side. Your data never leaves your computer. I've just added messaging to the UI to clarify this point. Thank you for making my app better!

"100% Private - All data conversion happens in your browser - We do not store your input or output files"

https://mapconverter.online/

**Reply by thany** [reply]
Then why do you need 10 cents for every conversion that you're doing on *my* computer?

(keyword is "need", I understand you *want* money - we all do)

**Reply by matt78704** [reply]
Why would I discuss my _needs_ with a _stranger_ on the internet?

**Reply by thany** [reply]
Why would you advertise your needs with a _bunch of stangers_ on the internet?

Look, I just want to know why it costs money if it's all local. You owe us an explanation if you want any chance of us using your service. But maybe that was a lie. Or a mistake. In which case you're welcome to admit it.

In either case, it seems like your seemingly innocent advertisement of a commercial product in a discussion of OSS, has already backfired.

**Reply by matt78704** [reply]
It was @tautologe who advertised his needs in the first post on this thread.

“I would like to import my Google Maps Saved Places to Organic Maps”

I am just providing a solution.

**Comment by SynapseScribe** [comment]
while i understand your standing point, this is not the right place to advertise your app for monetization

**Comment by biodranik** [comment]
Organic Maps can import Google Maps bookmarks now, please check https://github.com/orgs/organicmaps/discussions/928#discussioncomment-1049605

**Reply by thany** [reply]
Those are not the same as saved places. Maps has a couple of (different) lists of places, bookmarks are just one of them. The saved places are exported in CSV format only, for starters.
